coupons/models.py

```diff
@@ -37,6 +39,8 @@ def create_coupon(self, type, value, users=[], valid_until=None, prefix="", camp
         )
         if user_limit is not None:  # otherwise use default value of model
             coupon.user_limit = user_limit
+        if limit_per_user is not None: # otherwise use default value of model
```

Comment:
Why do this out here rather than passing to the instance constructor?  The current results in an extra save and I think splits up creation logic, causing you a bug (on conflict, the coupon created after will not have this extra stuff I think?)

Reply:
Updated to use `Coupon(...)` object instantiation with a single save instead of using `objects.create()` followed by a secondary save.